vllm_batch_inference/run.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+export HOME=$_CONDOR_SCRATCH_DIR
+export HF_HOME=$_CONDOR_SCRATCH_DIR/huggingface
+
+# If your job requests a single GPU, setting `CUDA_VISIBLE_DEVICES=0` ensures the system uses the correct device name and avoids errors. For multiple GPUs, set it accordingly (e.g., `0,1` for two GPUs).
+export CUDA_VISIBLE_DEVICES=0
```

Comment:
@iross is this necessary? I thought conventional wisdom was to not set `CUDA_VISIBLE_DEVICES` and let HTCondor manage it fully for single- and multi-GPU jobs?

Reply:
Looks like a `vllm` issue. Their [source](https://github.com/vllm-project/vllm/blob/cfbb8c930fcda6d97f0de3018bd3c51ab14b367c/vllm/platforms/cuda.py#L53) assumes `CUDA_VISIBLE_DEVICES` is an int. However, the NVIDIA [docs](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#cuda-environment-variables) say
> GPU identifiers are given as integer indices or as UUID strings.

and HTCondor sets them as UUID strings.

I vote to leave the script as it is but add in the comment this is a workaround for a `vllm` limitation, not something that is generally needed for GPU jobs.